datatypes_test.go

```diff
@@ -685,3 +685,37 @@ func testMaintenance(conveyMessage string, actualInfo MaintenanceInfo, maintenan
 		So(lastCheckTest.Maintenance, ShouldEqual, maintenance)
 	})
 }
+
+func TestCheckData_Serialization(t *testing.T) {
+	Convey("Test CheckData serialization works as expected", t, func() {
+		checkDataSet := []CheckData{
+			{State: StateNODATA},
+			{
+				State: StateNODATA,
+				Metrics: map[string]MetricState{
+					"123": {State: StateNODATA},
+					"321": {State: StateOK},
+					"345": {
+						EventTimestamp: 1504449789,
+						State:          StateNODATA,
+						Suppressed:     false,
+						Timestamp:      1504509380,
+						Values:         map[string]float64{"1": 1, "2": 2, "3": 3, "4": 4, "5": 5},
+					},
+				},
+			},
+			{Timestamp: 800, EventTimestamp: 0},
+			{Metrics: make(map[string]MetricState)},
+			{Maintenance: 1000},
+		}
+		for _, checkData := range checkDataSet {
+			serializedBytes, err := checkData.MarshalJSON()
+			So(err, ShouldEqual, nil)
```

Comment:
Let's add a serialization result check of both realizations

Reply:
Added